pgmpy/models/BayesianModel.py

```diff
@@ -738,6 +747,72 @@ def predict(self, data):
                 pred_values[k].append(v)
         return pd.DataFrame(pred_values, index=data.index)
 
+    def predict_probability(self, data):
```

Comment:
@raghavg7796 No. Since most of the code is common in both `predict` and `predict_probability`, I was suggesting something like:

```python
def _predict():
     # common code

def predict():
     # call _predict and do something with results

def predict_probability():
     # call _predict and do something with the values
```
or 
```python
def predict_probabilities():
     # compute the probabilities

def predict():
     # call predict_probabilities and use that to give the final predictions
```